homeassistant/components/yardian/switch.py

```diff
@@ -0,0 +1,100 @@
+"""Support for Yardian integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv, entity_platform
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import YardianUpdateCoordinator
+
+SERVICE_START_IRRIGATION = "start_irrigation"
+SERVICE_SCHEMA_START_IRRIGATION = {
+    vol.Required("duration"): cv.positive_int,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entry for a Yardian irrigation switches."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        YardianSwitch(
+            coordinator,
+            config_entry.as_dict(),
+            i,
+        )
+        for i in range(len(coordinator.data.zones))
+    )
+
+    platform = entity_platform.async_get_current_platform()
+    platform.async_register_entity_service(
+        SERVICE_START_IRRIGATION,
+        SERVICE_SCHEMA_START_IRRIGATION,
+        "async_turn_on",
+    )
+
+
+class YardianSwitch(CoordinatorEntity[YardianUpdateCoordinator], SwitchEntity):
```

Comment:
Please set `_attr_has_entity_name = True`

https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations